ql/src/java/org/apache/hadoop/hive/ql/ddl/table/create/CreateTableDesc.java

```diff
@@ -804,9 +801,18 @@ public Table toTable(HiveConf conf) throws HiveException {
       }
     }
 
-    if (getCols() != null) {
-      tbl.setFields(getCols());
+    Optional<List<FieldSchema>> cols = Optional.of(getCols());
```

Comment:
It seems like `getCols()` can return null based on the previous null check. This would fail here as well, unless wrap it using `Optional.ofNullable()`

Reply:
Thanks!
Done